### DISCUSSION TITLE
Is it possible to send push notification to user (not agent) when they receive message from an agent/admin?

### DISCUSSION THREAD

**Discussion by aekanshpartani** [chatwoot/chatwoot#2160] [Self hosted] [answered]
I got VAPID_KEYS generated and now I can see the configuration on setting page. But is there any way to send push notification to user on every new message?

**Comment by pranavrajs** [comment]
@aekanshpartani Who is the user here? Is it the agent or the end customer who is using the widget?

**Reply by WevrLabs** [reply]
I believe it refers to the visitor (freshchat does that, it notifies the visitor via push notifications if they march away from page before an agent responds)

**Reply by aekanshpartani** [reply]
> @aekanshpartani Who is the user here? Is it the agent or the end customer who is using the widget?
> 

I am taking about end user here

**Comment by sojan-official** [comment, marked as answer]
At the moment push notifications are limited to agents in chatwoot. We don't support push notifications for contacts.

**Reply by bilogic** [reply]
webpush is quite challenging to implement on a website with a different domain, especially if it also implements PWA, but how about having the chatwoot running on the help center page? This should make webpush for contacts much easier.

**Comment by Behtsa** [comment]
Are push notifications still limited to agents ???

**Comment by ibrahim-almahfooz** [comment]
Is this still a limitation or there is a way to implement it in the new versions ?
We checked the documents but no clear answer about it

Appreciate the support
